.github/workflows/CI.yml

```diff
@@ -9,15 +9,15 @@ jobs:
   Build:
     if: ${{ !contains(github.event.head_commit.message, '[skip ci]') }}
     runs-on: ${{ matrix.os }}
+    container: ${{ matrix.docker }}
     strategy:
       fail-fast: false
       matrix:
         os:
           # - ubuntu-22.04
           - ubuntu-20.04
           - windows-2022
-        node_version:
-          - 18
+        node_version: [10, 12, 14, 16, 18, 20, 22]
```

Comment:
I'd rather only smoke test the older versions. The scripts and tests are for the developers of this repository who have a controlled environment with access to the latest tools.

So, trying to run these for such old Node versions doesn't really help. 

For smoke testing, we can have setup-node install older node versions and just require zeromq to make sure it loads. Full testing will be done with the latest Node.

Reply:
I was not expecting CI to pass on these versions - I think we're on reasonably the same page.

Can you please fill in the blanks here?
- When you develop this package, it is okay to demand Node ___ and pnpm ___. (I think 22/9 is perfectly reasonable)
- When you consume this package, we expect you to be running Node ___ with package manager ___. (I think 18 / npm, pnpm, yarn berry)
- When you consume the types of this package, we expect TypeScript ___. (I think  >=4.7.4, following [the lead of typescript-eslint and DefinitelyTyped](https://typescript-eslint.io/users/dependency-versions/#typescript).

Reply:
These are good questions. I'd say whatever is defined in CI, is the minimum requirement for the developers, and we aim to support as many as wide Nodejs versions as possible. However, it is not always practical.

So to find the range of Nodejs versions that can consume this project, we can add the above-mentioned smoke tests.

We already have compact tests for TypeScript:
https://github.com/zeromq/zeromq.js/blob/828ed2854162ec6ec58473989dcda081209983f5/test/unit/typings-compatibility-test.ts#L27

Reply:
> I'd say whatever is defined in CI, is the minimum requirement for the developers, and we aim to support as many as wide Nodejs versions as possible. However, it is not always practical.
> So to find the range of Nodejs versions that can consume this project, we can add the above-mentioned smoke tests.

I'm not asking what versions can consume this library per CI - as of recently, this library was completely broken in *all* of its CI. I'm asking what versions should we fix existing tests and set up smoke tests for?

- I'm asking what versions, if broken, would we regard as a problem?
- Even that typings compatibility test is maybe not so clear. By setting the version to `4.x` I *think* that means only `typescript@4.9.5` is getting checked by CI, but your intention could have been that we support typescript as far back as 4.0. What version of TypeScript do you mean to target and why?